### PR TITLE
Deny warnings on CI (#329)

### LINKED ISSUE #329: Deny warnings on CI
subj. Just add `RUSTFLAGS='-D warnings'` and see how it goes perhaps?

### PR DESCRIPTION
**Pull request by DJMcNab**
Fixes #329.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,42 +1,47 @@
...
-  script:
-    - cargo gen-tests --verify
-    - cargo gen-syntax --verify
-    - cargo test
+    language: rust
+    rust: 1.31.1
+    script:
+        - cargo gen-tests --verify
+        - cargo gen-syntax --verify
+        - RUSTFLAGS='-D warnings' cargo test
```
Comment (matklad) [inline .travis.yml]: Why do we set `-D warnings` both here and via `env`? I'd prefer to just use `env`, but I was unable to figure out the proper yaml syntax for that :-)

Comment (DJMcNab) [reply .travis.yml]: I forgot to remove it :). Will fix now

**Comment by matklad** [comment]
yaml is not happy about the syntax: https://travis-ci.org/rust-analyzer/rust-analyzer/jobs/473282183#L414

**Comment by matklad** [comment]
Thanks!

**Pull request merged**